backend/src/lib.rs

```diff
@@ -983,7 +972,13 @@ pub extern "C" fn __quantum__qis__dumpmachine__body(location: *mut c_void) {
         let mut state = sim_state.borrow_mut();
 
         if !state.res.is_empty() {
-            println!("Global Results: {}", state.res);
+            OUTPUT.with(|output| {
+                let mut output = output.borrow_mut();
+                output
+                    .write_fmt(format_args!("Global Results: {}", state.res))
+                    .expect("Failed to write global results");
+                output.write_newline();
+            });
         }
         state.sim.dump();
```

Comment:
This produces a string that should go somewhere... we need to decide whether it ends up in the record or always goes to stdout.